--- platform.c
+++ platform.c
@@ -38,10 +38,12 @@
 
 #include <math.h>
 
+#ifdef _MSC_VER
 double round (float v)
 {
 	return floor(v+0.5f);
 }
+#endif
 
 // strndup() is not available on Windows
 char *strndup( const char *s1, size_t n)
--- platform.h
+++ platform.h
@@ -58,8 +58,10 @@
 #elif defined(_WIN32) || defined(_WIN64) 
     /* does not exist on windows */
     char * strndup( const char *s1, size_t n);
+#ifdef _MSC_VER
     double round (float v);
 #    pragma warning (disable: 4244) // suspend warnings
+#endif // _MSC_VER
 #endif // _WIN32 || _WIN64
 
 #ifdef __cplusplus
