--- ../poco/Util/CMakeLists.txt
+++ ../poco/Util/CMakeLists.txt
@@ -2,7 +2,6 @@
 
 set( BASE_SRCS
 src/AbstractConfiguration.cpp
-src/JSONConfiguration.cpp
 src/Application.cpp
 src/ConfigurationMapper.cpp
 src/ConfigurationView.cpp
@@ -25,7 +24,6 @@
 src/Subsystem.cpp
 src/SystemConfiguration.cpp
 src/Validator.cpp
-src/XMLConfiguration.cpp
 src/Timer.cpp
 src/TimerTask.cpp
 )
@@ -47,7 +45,7 @@
 set_target_properties( ${LIBNAME} 
	PROPERTIES
	VERSION ${SHARED_LIBRARY_VERSION} SOVERSION ${SHARED_LIBRARY_VERSION} )
-target_link_libraries( ${LIBNAME} PocoJSON PocoXML PocoFoundation)
+target_link_libraries( ${LIBNAME} PocoFoundation)
 
 install(
		 DIRECTORY include/Poco
